src/lib.rs

```diff
@@ -56,6 +61,74 @@ impl Reading {
     }
 }
 
+#[derive(Debug)]
+struct Output {
+    description: String,
+    error: Option<reqwest::Error>,
+    data: Option<Response>,
+    thread_id: ThreadId,
+    debug: bool
+  }
+
+impl Output {
+    pub fn new(
```

Comment:
I wonder if we could have one static instance of `Output` instead of having to call `new()` each time. That way it'd be a very lightweight operation both in execution time and in memory allocation. So it wouldn't be on the heap then, it'd be in a special section of memory called the [Block Starting Symbol (BSS)](https://en.wikipedia.org/wiki/.bss). These become global instances and are meant to be shared and used in different portions of an application.

Happy to discuss this more, just wanted to plant the idea.